File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        for i, a in enumerate(nums):
            # As List is in asc order, a greater than 0 indicates sum will never be 0
            if a > 0:
                break
            # Skip duplicates
            if i > 0 and nums[i - 1] == a: 
                continue
            l = i + 1
            r = len(nums) - 1
            while l < r:
                total = a + nums[l] + nums[r]
                if total > 0:
                    r = r - 1
                elif total < 0:
                    l = l + 1
                else:
                    res.append([a, nums[l], nums[r]])
                    # As match is found, Move both pointers inwards
                    l += 1
                    r -= 1
                    # Skip duplicates at the left pointer
                    while l < r and nums[l] == nums[l - 1]:
                        l += 1
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-1.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        if(nums.Length == 0) return false;
        Array.Sort(nums);
        int curr = nums[0];
        for(int i = 1; i < nums.Length; i++)
            if(nums[i] == curr)
                return true;
            else
                curr = nums[i];
        return false;
    }
}
